File: ShoppingM8-be/src/main/java/edu/pw/shoppingm8/list/invitation/api/dto/MyListInvitationsDto.java
package edu.pw.shoppingm8.list.invitation.api.dto;

import lombok.Builder;
import lombok.Value;

import java.util.Collection;

@Value
@Builder
public class MyListInvitationsDto {
    Collection<ListInvitationDto> sent;
    Collection<ListInvitationDto> received;
}

File: shoppingm8_fe/lib/list/invitation/dto/myListInvitationsDto.dart
import 'package:shoppingm8_fe/list/invitation/dto/listInvitationDto.dart';

class MyListInvitationsDto {
  final List<ListInvitationDto> sent;
  final List<ListInvitationDto> received;

  MyListInvitationsDto({this.sent, this.received});

  factory MyListInvitationsDto.fromJson(Map<String, dynamic> json) {
    List sentList = json['sent'] as List;
    List receivedList = json['received'] as List;
    return MyListInvitationsDto(
      sent: sentList.map((dto) => ListInvitationDto.fromJson(dto)).toList(),
      received: receivedList.map((dto) => ListInvitationDto.fromJson(dto)).toList()
    );
  }
}